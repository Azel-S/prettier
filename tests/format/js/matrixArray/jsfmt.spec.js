run_spec(__dirname, ["babel", "flow", "typescript"], {matrixArray: false});
run_spec(__dirname, ["babel", "flow", "typescript"], {matrixArray: true});
