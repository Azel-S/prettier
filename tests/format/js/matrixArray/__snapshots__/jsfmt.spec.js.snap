// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matrixArray.js - {"matrixArray":false} format 1`] = `
====================================options=====================================
matrixArray: false
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// Test file for array matrix formatting feature

// Single line array should remain unchanged
const testArray1 = [1, 2, 3, 4];

// Multiline array, already formatted
const testArray2 = [
    1, 2,
    3, 4
];

// Multiline array, needs reformatting
const testArray3 = [
    1, 2,
3, 4 
// Intentionally misaligned for testing
];

// Multiline array with more complex elements
const testArray4 = [
    [1, 2], [3, 4],
  [5, 6], [7, 8] // Intentionally misaligned for testing
];

=====================================output=====================================
// Test file for array matrix formatting feature

// Single line array should remain unchanged
const testArray1 = [1, 2, 3, 4];

// Multiline array, already formatted
const testArray2 = [1, 2, 3, 4];

// Multiline array, needs reformatting
const testArray3 = [
  1, 2, 3, 4,
  // Intentionally misaligned for testing
];

// Multiline array with more complex elements
const testArray4 = [
  [1, 2],
  [3, 4],
  [5, 6],
  [7, 8], // Intentionally misaligned for testing
];

================================================================================
`;

exports[`matrixArray.js - {"matrixArray":true} format 1`] = `
====================================options=====================================
matrixArray: true
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// Test file for array matrix formatting feature

// Single line array should remain unchanged
const testArray1 = [1, 2, 3, 4];

// Multiline array, already formatted
const testArray2 = [
    1, 2,
    3, 4
];

// Multiline array, needs reformatting
const testArray3 = [
    1, 2,
3, 4 
// Intentionally misaligned for testing
];

// Multiline array with more complex elements
const testArray4 = [
    [1, 2], [3, 4],
  [5, 6], [7, 8] // Intentionally misaligned for testing
];

=====================================output=====================================
// Test file for array matrix formatting feature

// Single line array should remain unchanged
const testArray1 = [1, 2, 3, 4];

// Multiline array, already formatted
const testArray2 = [
  1, 2,
  3, 4,
];

// Multiline array, needs reformatting
const testArray3 = [
  1, 2,
  3, 4,
  // Intentionally misaligned for testing
];

// Multiline array with more complex elements
const testArray4 = [
  [1, 2],
  [3, 4],
  [5, 6],
  [7, 8], // Intentionally misaligned for testing
];

================================================================================
`;
